src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -503,6 +503,44 @@ public string Encoding
         }
         string? _timezone;
 
+        /// <summary>
+        /// Determines the preferred PostgreSQL target server type.
+        /// </summary>
+        [Category("Connection")]
+        [Description("Determines the preferred PostgreSQL target server type.")]
+        [DisplayName("TargetSessionAttributes")]
+        [DefaultValue(TargetSessionAttributes.Any)]
+        [NpgsqlConnectionStringProperty]
+        public TargetSessionAttributes TargetSessionAttributes
+        {
+            get => _targetSessionAttributes;
+            set
+            {
+                _targetSessionAttributes = value;
+                SetValue(nameof(TargetSessionAttributes), value);
+            }
+        }
+        TargetSessionAttributes _targetSessionAttributes;
+
+        /// <summary>
+        /// Controls for how long the host's cached state will be considered as valid.
+        /// </summary>
+        [Category("Connection")]
+        [Description("Controls for how long the host's cached state will be considered as valid.")]
+        [DisplayName("HostRecheckSeconds")]
+        [DefaultValue(10)]
+        [NpgsqlConnectionStringProperty]
+        public int HostRecheckSeconds
+        {
+            get => _hostRecheckSeconds;
+            set
+            {
+                _hostRecheckSeconds = value;
```

Comment:
What happens if I set this to 0?

Reply:
With 0, it's only going to change if the host is going offline, or a new connector is open.

Reply:
Note that this should be meaningless (do nothing) on PostgreSQL 14, where state changes are supposed to be communicated to the client.

Reply:
Note offline decision to change 0 to mean "check every single time", since an infinite timeout means that an offline host never comes back.